numcodecs/vlen.pyx

```diff
@@ -28,6 +27,14 @@ cdef extern from "Python.h":
     int PyUnicode_Check(object text)
 
 
+# Hacky workaround for lack of `const object[:]`.
+# Appears hiding `object` in a fused type works.
+# xref: https://github.com/cython/cython/issues/2485
+# xref: https://github.com/scipy/scipy/pull/18192#pullrequestreview-1359581611
+ctypedef fused object_fused_t:
+    object
+    uintptr_t
```

Comment:
This is the crux of the workaround

For fusing, used a `uintptr_t`. As `object` is effectively `PyObject*`, fusing with a generic pointer/address type seemed most appropriate

Note this differs from the SciPy approach where they used `double`. Though it shouldn't matter much what the other type is as it is unused

Reply:
Well as evidenced by CI, this is not enough. Think what we need to do is get the fused types inside some kind of function signature. Then it might work

However looking at the code here. There are multiple places this would need to go

Wonder if the better path would be...

1. Move most of this logic into pure Python (Cython is only needed for a small portion of this code)
2. Refactor the code to reuse common bits across cases
3. Have a small Cython function that deals with the bits that need to be lower-level